static_report/src/components/shared/Charts/HistogramChart.tsx

```diff
@@ -0,0 +1,188 @@
+import {
+  ChartOptions,
+  Chart as ChartJS,
+  BarElement,
+  Tooltip,
+  ChartData,
+  ScaleOptionsByType,
+  CartesianScaleTypeRegistry,
+  TimeSeriesScale,
+  CategoryScale,
+  LinearScale,
+} from 'chart.js';
+import { Chart } from 'react-chartjs-2';
+import { ColumnSchema, Histogram } from '../../../sdlc/single-report-schema';
+import {
+  formatAsAbbreviatedNumber,
+  formatIntervalMinMax,
+} from '../../../utils/formatters';
+import {
+  DATE_RANGE,
+  TEXTLENGTH,
+  VALUE_RANGE,
+} from '../ColumnCard/ColumnTypeDetail/constants';
+
+import 'chartjs-adapter-date-fns';
+import { DeepPartial } from 'chart.js/types/utils';
+
+ChartJS.register(
+  TimeSeriesScale,
+  LinearScale,
+  CategoryScale,
+  BarElement,
+  Tooltip,
+);
+/**
+ * Histogram Chart that can display generic data types such as Numeric, Datetime, Integer
+ * X: The Min/Max of the labels range is the scaled width of charting area
+ * Y: The Min/Max of the counts range is the scaled height of charting area
+ * Counts: Abbreviated based on K, Mn, Bn, Tr (see formatters)
+ */
+//Note: min/max represents the bin edge min/max
+type ScaleTypeConfig = DeepPartial<
+  ScaleOptionsByType<keyof CartesianScaleTypeRegistry>
+>;
+type Props = {
+  data: Pick<ColumnSchema, 'total' | 'type' | 'histogram' | 'min' | 'max'>;
+};
+export function HistogramChart({
+  data: { histogram, type, total, min, max },
+}: Props) {
+  const { counts, bin_edges } = histogram as Histogram;
```

Comment:
Is not necessary to change if you feel it using `bin_edges` is okay.

```suggestion
  const { counts, bin_edges as binEdges } = histogram as Histogram;
```

Reply:
+1 to keep consistent, but i think should be `bin_edges: binEdges` (assuming that's what you meant for renaming a destructured property)

Reply:
yes, i'm thinking about import syntax xd